File: WIP - Progetto/app/src/main/java/it/wip/database/model/Story.kt
package it.wip.database.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(tableName = "story",
        foreignKeys = [
            ForeignKey(
                entity = User::class,
                parentColumns = ["id"],
                childColumns = ["user"],
                onUpdate = ForeignKey.CASCADE,
                onDelete = ForeignKey.CASCADE
            )
        ]
)
data class Story(
    @PrimaryKey(autoGenerate = true)
    val id: Int,
    @ColumnInfo(name = "story_name")
    val storyName: String,
    @ColumnInfo(name = "created_on")
    val createdOn: String,
    val user: Int
)

File: wip_flutter/lib/database/model/story.dart
class Story {

  final int id;
  final String storyName;
  final String createdOn;
  final int user;

  Story({
    required this.id,
    required this.storyName,
    required this.createdOn,
    required this.user});

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'story_name': storyName,
      'created_on': createdOn,
      'user': user
    };
  }

}